server/form-pages/apply/add-documents/attachDocuments.ts

```diff
@@ -68,23 +72,28 @@ export default class AttachDocuments implements TasklistPage {
         'N/A': 'No documents attached',
       }
     }
-
     this.body.selectedDocuments.forEach(d => {
       response[d.fileName] = d.description || 'No description'
     })
-
     return response
   }
 
   errors() {
     const errors: TaskListErrors<this> = {}
-
-    this.body.selectedDocuments.forEach(document => {
-      if (!document.description) {
-        errors[`selectedDocuments_${document.id}`] =
-          `You must enter a description for the document '${document.fileName}'`
-      }
-    })
+    // Check number of docs selected is docs is OK.
+    if (this.body.selectedDocuments.length > maxDocumentsToUpload) {
+      errors[`selectedDocuments_${this.body.selectedDocuments[maxDocumentsToUpload].id}`] =
+        `You can only select ${maxDocumentsToUpload} documents, remove ${this.body.selectedDocuments.length - maxDocumentsToUpload} to continue.`
+    }
+    if (!Object.keys(errors).length) {
+      // Only check descriptions if number of docs is OK so that the user doesn't waste their time adding descriptions.
```

Comment:
A test to cover this behaviour would be nice, and remove the need for this comment -- it could be covered in the test to show errors for too many docs. When using the `buildList` factory method, you can still provide default params, i.e. `documentFactory.buildList(6, { description: undefined })`

Reply:
Fair enough.